src/redux/store.ts

```diff
@@ -0,0 +1,8 @@
+import { configureStore } from "@reduxjs/toolkit";
+import { reducer } from "./slices/index";
+
+export const store = configureStore({ reducer });
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<typeof store.getState>;
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
+export type AppDispatch = typeof store.dispatch;
```

Comment:
Had to add redux, it was too difficult to manage the state from the search bar now that components are so far apart.